pkg/services/searchV2/object/dashboard.go

```diff
@@ -0,0 +1,71 @@
+package object
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/searchV2/dslookup"
+	"github.com/grafana/grafana/pkg/services/searchV2/extract"
+	"github.com/grafana/grafana/pkg/services/store/object"
+)
+
+func NewDashboardSummaryBuilder(lookup dslookup.DatasourceLookup) object.ObjectSummaryBuilder {
+	return func(obj *object.RawObject) (object.ObjectSummary, error) {
+		summary := object.ObjectSummary{
+			Labels: make(map[string]string),
+			Fields: make(map[string]interface{}),
+		}
+		stream := bytes.NewBuffer(obj.Body)
+		dash, err := extract.ReadDashboard(stream, lookup)
+		if err != nil {
+			summary.Error = &object.ObjectErrorInfo{
+				Code:    500, // generic bad error
+				Message: err.Error(),
+			}
+			return summary, err
+		}
+
+		refs := object.NewReferenceAccumulator()
+		url := fmt.Sprintf("/d/%s/%s", obj.UID, models.SlugifyTitle(dash.Title))
+		summary.Name = dash.Title
+		summary.Description = dash.Description
+		summary.URL = url
+		for _, v := range dash.Tags {
+			summary.Labels[v] = ""
+		}
+		if len(dash.TemplateVars) > 0 {
+			summary.Fields["hasTemplateVars"] = true
+		}
+
+		for _, panel := range dash.Panels {
+			refP := object.NewReferenceAccumulator()
+			p := object.ObjectSummary{
+				UID:  obj.UID + "#" + strconv.FormatInt(panel.ID, 10),
+				Kind: "panel",
+			}
+			p.Name = panel.Title
+			p.Description = panel.Description
+			p.URL = fmt.Sprintf("%s?viewPanel=%d", url, panel.ID)
+			p.Fields = make(map[string]interface{}, 0)
+
+			refP.Add("panel", panel.Type, "")
+			for _, v := range panel.Datasource {
+				refs.Add("ds", v.Type, v.UID) // dashboard refs
+				refP.Add("ds", v.Type, v.UID) // panel refs
+			}
+
+			for _, v := range panel.Transformer {
+				refP.Add("transformer", v, "")
```

Comment:
hmmm 🤔 I am not sure about treating transformers as entities. Don't they belong to "fields"?

Reply:
yeah, I'd say this applies also to panels. I don't like the fact that we will have a set of references for which the entity API methods (such as `entityAPI.get(reference)`) won't work. This leak in abstraction can lead to many `ifs` in code, examples:

1. I have a dashboard summary with three "external references", one to a panel and a transformer, and now I want to retrieve the full entity graph. 

```
{
  "kind": "dashboard",
  "uid": 100,
  "references": [
    {
      "id": "100#1",
      "type": "panel",
      "kind": "abc"
    },
    {
      "id": "",
      "type": "transform",
      "kind": "join"
    },
    {
      "id": "x1234d",
      "type": "datasource",
      "kind": "mongo"
    }
  ]
}
```

I can't simply do `entityAPI.batchGet(dashboard.references)`, because some of the references intentionally don't exist

2. permissions - I now need to differentiate between different "types" of kinds:
 -  "invalid", 
 -  "virtual-but-always-accessible" (transforms)
 -  "virtual-but-inherited" (panels)
 - "standard"

3. Navigation - some entities have a `storage/{uid}` page, some don't

-----

I am not sure if this is a big deal - but its also something worth considering. Maybe having a `NestedObject` concept that's identified by a `NestedObjectId` rather then the entity reference would be helpful